src/ripple/rpc/DeliveredAmount.h

```diff
@@ -63,13 +66,12 @@ insertDeliveredAmount(
     std::shared_ptr<Transaction>,
     TxMeta const&);
 
-void
-insertDeliveredAmount(
-    rpc::v1::CurrencyAmount& proto,
-    Context&,
-    std::shared_ptr<Transaction>,
-    TxMeta const&);
-
+std::optional<STAmount>
+getDeliveredAmount(
```

Comment:
I do use the `getDeliveredAmount` function elsewhere. I could just change the public function to take the ledgerIndex in as a parameter, instead of this callable.
Curious though, `std::function` might allocate even when passed by const&? If we changed this to a template, how does that stop allocation? What type would the template parameter resolve to, that avoids allocation?

Reply:
Lambdas can have different sizes - it depends on what the lambda captures. `std::function` has a fixed sized. When we construct a `std::function` from a lambda, if the lambda is small,
it won't allocate. But if the lambda is bigger, it needs to allocate (much the same way a string may or may not allocate).

Passing my reference doesn't stop the allocation. If it allocates, it'll happen when the `std::function` is constructed from a lambda. If we make the parameter a template, then the parameter type is same type as the lambda. There's no need for type erasure and no need for allocation.

I.e.
```
void mayAllocateForParam(std::function<void ()> const& f){f();};

template<class T>
void willNotAllocateForParam(T&& f){f()};

auto lambdaThatCapturesABunchOfStuff = [=]{...};
mayAllocateForParam(lambdaThatCapturesABunchOfStuff);
willNotAllocateForParam(lambdaThatCapturesABunchOfStuff);
```

If that doesn't make sense, I'm happy to hop on a call and do my best to explain.

Reply:
No need for a call, makes sense to me. I'll change to a template